lib/ansible/plugins/action/__init__.py

```diff
@@ -680,6 +680,12 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
         display.debug("done with _execute_module (%s, %s)" % (module_name, module_args))
         return data
 
+    def _remove_internal_keys(self, data):
+        for key in list(data.keys()):
+            if key.startswith('_ansible_') or key in C.INTERNAL_RESULT_KEYS:
```

Comment:
I heard you didn't like the ``` list(data.keys())``` idiom.  Since you're really filtering a list here, you can do it with a list comprehension instead:
``` python
for key in  [k for k in data.keys() if key.startswith('_ansible_') or key in C.INTERNAL_RESULT_KEYS ]:
    display.warning([...])
    del data[key]
```
With such a long conditional, I like to break the filtering over one or more statements like these examples:

``` python
badkeys = [k for k in data.keys() if key.startswith('_ansible_') or key in C.INTERNAL_RESULT_KEYS ]
for key in badkeys:
    [...]

# Generator expression so we don't create an extra list (sometimes slower but always uses less memory)
badkeys = (k for k in data.keys() if key.startswith('_ansible_'))
# list comprehension so that we don't delete keys from the dict while a generator is still looping through them
badkeys = [k for k in badkeys if key in C.INTERNAL_RESULT_KEYS]
for key in badkeys:
    [...]
```

Choose whichever idiom is the most readable to you.